File: Python/CIP/04 M5 - Leetcode 560 - Subarray Sum Equals K/Leetcode 560 - Subarray Sum Equals K/(O(n^2)/560-external.py

# https://leetcode.com/problems/subarray-sum-equals-k/discuss/102144/TLE-for-cumulative-sum-in-python
# Time Limit Exceeded	

class Solution(object):
    def subarraySum(self, nums, k):
        if not nums:
            return 0
        count = 0
        nLen = len(nums)
        for i in range(nLen):
            last = 0
            for j in range(i, nLen):
                last = last + nums[j]
                if last == k:
                    count += 1
        return count

File: Python/CIP/04 M5 - Leetcode 560 - Subarray Sum Equals K/Leetcode 560 - Subarray Sum Equals K/(O(n^2)/560-external.js

// https://leetcode.com/problems/subarray-sum-equals-k/discuss/820048/JavaScript-simplest-brute-force-solution-O(N-2)

var subarraySum = function(nums, k) {
    let count = 0;
    
    for (let i = 0; i < nums.length; i++) {
        let sum = 0;
        
        for (j = i; j < nums.length; j++) {
            sum += nums[j];
            
            if (sum === k) {
                count++;
            }
        }
    }
    
    return count;
};
